sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala

```diff
@@ -169,10 +162,33 @@ case class AdaptiveSparkPlanExec(
     optimized
   }
 
+  private def checkDistribution(
+      newPlan: SparkPlan,
+      originPlan: SparkPlan,
+      isFinalStage: Boolean,
+      ruleName: String): SparkPlan = {
```

Comment:
this method is used by `optimizeQueryStage` and `optimizeSkewedJoin` for checking final stage distribution